lib/drivers/onetouch/oneTouchUltra2.js

```diff
@@ -179,6 +179,110 @@ module.exports = function (config) {
 
     var toMatch = hpattern.exec(header);
 
+    var parseData = function() {
+      var count = 0;
+
+      async.whilst(
+        function () { return (splits.length - 2) < nrecords; },
+        function (callback) {
+          // we still need to read more data
+          debug('Reading more data..');
+          count++;
+          progress((count / nrecords) * 100);
+          listenForPacket(18000, progress, function(err, result) {
+            if (err) {
+              return callback(err);
+            }
+            var bytes = byteConcat(data.bytes, result.bytes.slice(1));
+            _.assign(data, buildPacket(bytes));
+            lines = String.fromCharCode.apply(null, data.bytes);
+            splits = lines.split('\n');
+            return callback();
+          });
+        },
+        function (err, n) {
+          if (err) {
+            return cb(err);
+          }
+
+          // get data from lines
+          var dpattern = /P "(\w+)","(\d{2}\/\d{2}\/\d{2})","(\d{2}:\d{2}:\d{2})   ","(..\d{3} )","(.)","(\d{2})",\W00\W(.+)/;
```

Comment:
What are the extra `..` characters in `(..\d{3} )`? `(.)` is very inclusive. Can it include whitespace characters?

Reply:
I copied & pasted this regex from another section in the Ultra2 driver - we've been using it for years without any problems. TBH it's one of those where I can't even remember who actually wrote the code.

Looking at my regex cheatsheet `(.)` can include any character except for newlines.